code/core/src/components/components/utils/createCopyToClipboardFunction.ts

```diff
@@ -0,0 +1,21 @@
+export function createCopyToClipboardFunction() {
+  if (navigator?.clipboard) {
+    return (text: string) => navigator.clipboard.writeText(text);
+  }
+
+  return async (text: string) => {
```

Comment:
nit: You could make these two functions top-level `function`s instead of arrow functions, so that no matter how often this file is imported at the top-level of new ES modules like you new in `mdx.tsx`, the same function declarations are reused instead of new ones being held in memory. Function declarations are hoisted and arrow functions aren't.